### PR TITLE
Move the implementation of server side ExplainerApi to a dedicated class

### PR DESCRIPTION
**Pull request by shtukas**
In order to implement the user update of atoms, I need to move the implementation of trait `ApiController` from the `ApiController` to a dedicated class.

### CODE REVIEW

File: explainer-server/app/controllers/ApiController.scala [hunk 1]
```diff
@@ -30,26 +30,15 @@ object AutowireServer extends autowire.Server[Js.Value, Reader, Writer]{
   def write[Result: Writer](r: Result) = upickle.default.writeJs(r)
 }
 
-class ApiController @Inject() (config: Config, previewAtomPublisher: PreviewAtomPublisher,
-  val publicSettingsService: PublicSettingsService,
-  val liveAtomPublisher: LiveAtomPublisher) extends Controller with ExplainerApi with AuthActions  {
+class ExplainerApiC(
```
Comment (philmcmahon) [inline explainer-server/app/controllers/ApiController.scala]: ExplainerApi

**Comment by philmcmahon** [comment]
👍

File: explainer-server/app/controllers/ApiController.scala (hunk 1 above)
Comment (shtukas) [inline explainer-server/app/controllers/ApiController.scala]: `illegal cyclic reference involving class ExplainerApi` :)

Comment (shtukas) [inline explainer-server/app/controllers/ApiController.scala]: As discussed going for ExplainerApiImpl

**Pull request merged**